config/configauth/pdatacontext.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type ctxKey struct{}
+
+type AuthContext interface {
+	Equal(other interface{}) bool
+	GetAttribute(attrName string) interface{}
+}
+
+func InjectAuthContext(ctx context.Context, attrs map[interface{}]interface{}) context.Context {
+	ac := &authC{delegate: attrs}
+	return context.WithValue(ctx, ctxKey{}, ac)
+}
+
+func ExtractAuthContext(ctx context.Context) (AuthContext, bool) {
+	ac, ok := ctx.Value(ctxKey{}).(*authC)
+	if !ok {
+		return nil, false
+	}
+	return ac, true
+}
+
+func InjectPDataContext(pda pdata.PDataContext, ac AuthContext) {
+	pda.Set(ctxKey{}, ac)
+}
+
+func ExtractPDataContext(pda pdata.PDataContext) AuthContext {
+	return pda.Get(ctxKey{}).(AuthContext)
+}
+
+type authC struct {
```

Comment:
Do we need `authC` in this package? I think we can move it to `oidcauth`, right? `InjectAuthContext` can just take `AuthContext` parameter instead of attrs.
Do we gain anything by having the concrete `authC` implementation in this package?